app/scripts/web/utils/pdf_generator.py

```
@@ -0,0 +1,260 @@
+    # Save & update FIFO index
+    safe_name    = re.sub(r"[^\w\-]", "_", student)[:40]
+    pdf_filename = f"{run_id}_{safe_name}.pdf"
+    pdf_path     = REPORTS_DIR / pdf_filename
+    try:
+        pdf.output(str(pdf_path))
+    except Exception:
+        return None
+
+    try:
+        index: list[dict] = json.loads(REPORTS_INDEX.read_text()) if REPORTS_INDEX.exists() else []
+    except Exception:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)